code/addons/docs/src/blocks/getStoryHref.ts

```diff
@@ -0,0 +1,23 @@
+export const getStoryHref = (storyId: string, additionalParams: Record<string, string> = {}) => {
+  const baseUrl = globalThis.PREVIEW_URL || 'iframe.html';
+  const [url, paramsStr] = baseUrl.split('?');
+  const params = {
+    ...(paramsStr ? parseQuery(paramsStr) : {}),
+    ...additionalParams,
+    id: storyId,
+  };
+  return `${url}?${Object.entries(params)
+    .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
+    .join('&')}`;
+};
+
+function parseQuery(queryString: string) {
```

Comment:
Any particular reason we don't use `Object.fromEntries(new URLSearchParams(queryString))`?